src/main/java/dinkplugin/DinkPlugin.java

```diff
@@ -76,6 +94,33 @@ DinkPluginConfig provideConfig(ConfigManager configManager) {
         return configManager.getConfig(DinkPluginConfig.class);
     }
 
+    @Subscribe
+    public void onConfigChanged(ConfigChanged event) {
```

Comment:
should this logic be moved to a separate class

Reply:
I think this is very much up to you if it makes sense from a Java perspective to have a specific class be responsible for this. I could see a "SettingValidator" class or something have the responsibility for checking & posting warnings, while this class would just post events into it but up to you